nasa/harmony
# DAS-2044 - Update Swath Projector image and environment variables.

## Jira Issue ID

DAS-2044

## Description

This PR updates Harmony to use the [new open-source Docker image for the Swath Projector](https://github.com/nasa/harmony-swath-projector/pkgs/container/harmony-swath-projector). I also have taken the chance to update the name to "Swath Projector" everywhere, as the "swot-repr" name was deprecated a while back. Those environment variable changes are a little more pesky, but I'm also about to push a PR to the CI/CD repo updating the Terraform code there, too.

For what it's worth, I think the other projects being migrated mostly have variables names that are still accurate (like HyBIG and MaskFill), so those migrations should just be a case of swapping out the Docker image location.

## Local Test Steps

* Pull this branch.
* Pull the [new open-source Docker image](https://github.com/nasa/harmony-swath-projector/pkgs/container/harmony-swath-projector) locally: `docker pull ghcr.io/nasa/harmony-swath-projector:latest`
* Update your `LOCALLY_DEPLOYED_SERVICES` to replace "swot-repr" (or "swath-reproject") with "swath-projector".
* Rebuild the versions of the local Harmony images you are using: `nvm use && npm i && npm run build-all`
* Start a local Harmony instance: `./bin/bootstrap-harmony`.
* Start up a Jupyter notebook server and run through the [Swath Projector regression test suite](https://github.com/nasa/harmony-regression-tests/tree/main/test/swath-projector) after setting `harmony_host_url='http://localhost:3000` in the notebook. The tests should run and they should still pass.

## PR Acceptance Checklist
* ~~Documentation updated (if needed)~~
